_pytest/python.py

```diff
@@ -594,7 +603,10 @@ def setup(self):
 
     def _prunetraceback(self, excinfo):
         if hasattr(self, '_obj') and not self.config.option.fulltrace:
-            code = py.code.Code(self.obj)
```

Comment:
did you try just ``code = py.code.Code(get_real_func(self.obj))``?  Does it break any tests?